src/rust/iced-x86-py/src/instruction.rs

```diff
@@ -166,6 +169,31 @@ impl Instruction {
 		Self { instr: iced_x86::Instruction::default() }
 	}
 
+	/// Set the internal state with the given unpickled state.
+	///
+	/// Args:
+	///     state (Any): unpickled state
+	#[pyo3(text_signature = "($self, state, /)")]
+	fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+		match state.extract::<&PyBytes>(py) {
```

Comment:
Can the input be `PyByteArray`? If so we need to check for it here too.

Reply:
Input should be what we have used in `__getstate__` so I don't think it's ok to accept bytearray.